portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,11 +36,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
  @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      long timestamp = System.currentTimeMillis();
       String textShows = request.getParameter("text-input");
       String[] arrShows = textShows.split("\\s*,\\s*");
       for(String tvShow : arrShows){
           shows.add(tvShow);
+          Entity taskEntity = new Entity("Shows");
```

Comment:
So each REST operation should probably correspond to 1 datastore transaction.  Here you actually have 1 POST leading to many transactions.  The issue there is if I click "submit" some of the inserts into Shows might succeed bug some might fail.  So you have two options here:

1) Make the insert atomic by doing the multiple inserts in a transaction: https://cloud.google.com/datastore/docs/concepts/transactions#datastore-datastore-transactional-update-java

2) Change the UI s.t. each form post corresponds to a HTTP post request, s.t. you're inserting a single show per request.  You might 

I think option #2 is probably what you want, and it would be more RESTful.  

--------
More detail that probably is too much for the starter projects:

Here's a case study example.  Let's say you're designing a webpage to reserve tickets.  Let's say you can multiple Passenger per Reservation entity.  You could design it such that you have 1 form that creates the Reservation and all the passengers and does the purchasing at the end with a single POST.  But really probably you'll be piecing together the Reservation with an initial POST with subsequent PATCH requests.    And you'll probably be adding passengers one at a time (with its separate POST requests).  To finally reserve the reservation it might just be a PATCH to update the state to "RESERVED" or something.

Reply:
Did option 2.  Makes sense to only have one transaction per post request.